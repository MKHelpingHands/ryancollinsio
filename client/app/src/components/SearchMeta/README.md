## SearchMeta Component
A component that ...

### Example

```js
<SearchMeta />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
