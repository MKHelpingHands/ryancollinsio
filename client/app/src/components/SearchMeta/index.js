import React, { PropTypes } from 'react';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';

const SearchMeta = ({
  searchTerm,
  tags,
  array,
}) => (
  <Box>
    {array && array.length > 0 && searchTerm &&
      <Heading align="center" tag="h3">
        {`${array.length} items found for: ${searchTerm}`}
      </Heading>
    }
    {tags && tags.length > 0 &&
      <Heading align="center" tag="h3">
        {`${array.length} items found for: ${tags.map((tag) => tag.title).join(', ')}`}
      </Heading>
    }
    {searchTerm && !array || array && !array.length > 0 &&
      <Heading align="center" tag="h3">
        None Found
      </Heading>
    }
  </Box>
);

SearchMeta.propTypes = {
  searchTerm: PropTypes.string,
  array: PropTypes.array,
};

export default SearchMeta;
