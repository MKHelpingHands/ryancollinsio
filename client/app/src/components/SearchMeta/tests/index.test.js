import SearchMeta from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SearchMeta />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SearchMeta array={[]} searchTerm="Hello World" />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
