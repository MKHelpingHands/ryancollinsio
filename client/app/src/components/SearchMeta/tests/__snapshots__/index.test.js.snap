exports[`<SearchMeta /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad="none"
  responsive={true}
  tag="div">
  <Heading
    align="center"
    tag="h3">
    None Found
  </Heading>
</Box>
`;
